okhttp/src/test/java/io/grpc/transport/okhttp/OkHttpClientTransportTest.java

```diff
@@ -1004,6 +1056,87 @@ public void ping_failsIfTransportFails() throws Exception {
         ((StatusException) callback.failureCause).getStatus().getCode());
   }
 
+  @Test
+  public void writeBeforeConnected() throws Exception {
+    initTransport(3, new ConnectedCallback(true));
+    final String message = "Hello Server";
+    MockStreamListener listener = new MockStreamListener();
+    OkHttpClientStream stream = clientTransport.newStream(method, new Metadata.Headers(), listener);
+    InputStream input = new ByteArrayInputStream(message.getBytes(UTF_8));
+    stream.writeMessage(input);
+    stream.flush();
+    // The message should be queued.
+    verifyNoMoreInteractions(frameWriter);
+
+    connectedCallback.connected();
```

Comment:
I don't understand why connectedCallback is necessary. Like here just commenting out this line seems like it should work, because we have a timeout on the verify() and so it will wait for the frameWriter to be called.